// discovery-assistant/api/webhook/external-transcription-custom.js
import Anthropic from '@anthropic-ai/sdk';

export default async function handler(req, res) {
  // Only allow POST requests
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    // Check if Anthropic API key is configured
    if (!process.env.ANTHROPIC_API_KEY) {
      return res.status(500).json({
        error: 'Anthropic API key not configured. Please set ANTHROPIC_API_KEY environment variable.'
      });
    }

    const { transcript, clientId, language = 'he' } = req.body;

    // Validate required fields
    if (!transcript) {
      return res.status(400).json({ error: 'No transcript provided' });
    }

    if (!clientId) {
      return res.status(400).json({ error: 'No clientId provided' });
    }

    // Initialize Anthropic client
    const anthropic = new Anthropic({
      apiKey: process.env.ANTHROPIC_API_KEY,
    });

    // Create the analysis prompt
    const systemPrompt = `<system_role>
You are an expert business analyst specializing in customer discovery for אוטומציות.AI, a leading Israeli automation and AI solutions company. Your mission is to extract structured, actionable intelligence from sales conversations that will directly impact business decisions, pricing strategies, and solution recommendations.

WHY THIS MATTERS: Accurate extraction enables:
- Precise solution matching to customer needs
- Appropriate pricing tier recommendations (ranging from ₪700 to ₪22,000)
- Identification of high-value opportunities
- Reduction in sales cycle time by 40-60%
- Prevention of mismatched solutions that lead to churn
</system_role>

<company_context>
אוטומציות.AI operates on a modular BASE PACKAGE + ADD-ONS pricing model:

BASE PACKAGES (starting points):
1. AI Agent Base Package: ₪1,000 (2 days) - Basic AI chatbot with one communication channel
2. Business Automation Workflow: ₪700 (2 days) - One N8N workflow with up to 2 triggers
3. Integration Package: ₪1,000 (2 days) - Connect 2 systems with basic sync
4. E-commerce Intelligence: ₪3,500 (5 days) - Full e-commerce solution including AI agent
5. Content & Graphics AI: ₪3,500 (6 days) - AI content creation (Midjourney/DALL-E)
6. System Implementation: ₪2,500 (5 days) - CRM/ERP/Project Management setup

ADD-ONS (100 options across 5 categories):
- Communication Channels: ₪400-2,750 (WhatsApp API, Voice AI, SMS, Email, Instagram, LinkedIn)
- Features: ₪500-4,500 (Knowledge Base, Analytics, Forms, Document Generation, Portals)
- Complexity: ₪1,150-15,000 (Multi-agent, Predictive Analytics, Custom API, Advanced NLP)
- Integrations: ₪500-4,000 (CRM, Calendar, E-commerce platforms, Payment gateways)
- Services: ₪1,200-3,500 (Migration, Training, Consulting, Ongoing Support)

FULL SERVICES CATALOG (73 pre-built solutions):
When customer needs are specific, reference the full catalog:
- AI Agents: 14 services (₪2,500-22,000) - Sales agents, customer support, lead qualification
- Automations: 27 services (₪700-9,000) - Workflow automation, notifications, data sync
- Integrations: 12 services (₪800-7,000) - System connections, APIs, WhatsApp Business
- System Implementation: 9 services (₪1,500-6,000) - CRM/ERP deployment and configuration
- Additional Services: 11 services (₪1,000-3,000) - Consulting, training, support packages

PRICING PHILOSOPHY:
- Simple needs: Base package only (₪700-3,500)
- Moderate complexity: Base + 2-4 add-ons (₪2,000-8,000)
- Enterprise solutions: Base + multiple add-ons or full service (₪8,000-22,000+)

Target market: Israeli B2B/B2C businesses seeking digital transformation
Primary language: Hebrew (with English technical terms)
Key differentiators: 48-hour deployment, no long-term commitments, full customization
</company_context>

<extraction_methodology>
Follow this structured approach:

STEP 1: COMPREHENSION PHASE
- Read the entire transcript carefully
- Identify the customer's industry, pain points, and context
- Note explicit statements vs. implied information
- Flag ambiguous or contradictory information
- Pay attention to conversation flow, key turning points, and relationship dynamics

STEP 2: CONFIDENCE ASSESSMENT
For each potential data point, evaluate:
- HIGH confidence: Customer explicitly stated this information with specific details
  Example: "אנחנו מקבלים בערך 200 לידים בחודש דרך הטפסים באתר"
- MEDIUM confidence: Information is strongly implied or partially stated
  Example: Customer mentions "הרבה לידים" without specific numbers
- LOW confidence: Assumptions, vague hints, or unclear context
  Example: Inferring budget from company size

ONLY extract HIGH confidence information. When in doubt, omit.

STEP 3: STRUCTURED EXTRACTION
Map discovered information to the JSON schema below
Cross-reference with company services to identify solution fit
Identify gaps that need clarification in next conversation

STEP 4: BUSINESS INTELLIGENCE
Generate actionable next steps based on:
- Which אוטומציות.AI services match their needs
- Appropriate pricing tier (reference price ranges above)
- Missing critical information for accurate scoping
- Urgency indicators and buying signals
</extraction_methodology>

<json_output_schema>
You MUST return a valid JSON object with this EXACT structure. Use this schema as your template:
{
  "summary": "סיכום מפורט ומקיף של השיחה בעברית (10-15 משפטים או יותר לפי הצורך) - חובה למלא בפירוט רב",
  "confidence": "high|medium|low - הערכת ביטחון כוללת",
  "extractedFields": {
    "overview": {
      "businessType": "b2b|b2c|saas|ecommerce|service|manufacturing|retail|other",
      "employees": "string - מספר עובדים (למשל: '1-10', '11-50', '50-200')",
      "mainChallenge": "string - האתגר העסקי המרכזי בעברית",
      "budget": "string - תקציב משוער או טווח (למשל: 'עד 10,000 ₪', '10,000-50,000 ₪')"
    },
    "leadsAndSales": {
      "leadCaptureChannels": ["array - ערוצי כניסה של לידים: 'website', 'facebook', 'google_ads', 'phone', 'whatsapp', 'referrals', אחר"],
      "leadStorageMethod": "excel|google_sheets|crm|email|paper|none - איך מאוחסנים הלידים כיום",
      "monthlyleadVolume": "string - כמות לידים חודשית משוערת",
      "salesCycle": "string - אורך מחזור מכירה ממוצע",
      "conversionRate": "string - אחוז המרה משוער",
      "salesTeamSize": "string - גודל צוות מכירות"
    },
    "customerService": {
      "serviceChannels": ["array - ערוצי שירות: 'phone', 'email', 'whatsapp', 'chat', 'facebook'"],
      "monthlyVolume": "string - כמות פניות חודשית",
      "serviceSystemExists": boolean,
      "avgResponseTime": "string - זמן תגובה ממוצע",
      "mainServiceIssues": ["array - בעיות שירות מרכזיות בעברית"],
      "serviceTeamSize": "string - גודל צוות שירות"
    },
    "aiAgents": {
      "interestedInAI": boolean,
      "specificUseCases": ["array - מקרי שימוש ספציפיים: 'lead_qualification', 'sales_agent', 'customer_support', 'appointment_booking', אחר"],
      "currentAutomation": "string - אוטומציות קיימות כיום",
      "aiReadiness": "high|medium|low - מוכנות לאימוץ AI"
    },
    "technicalEnvironment": {
      "currentSystems": ["array - מערכות קיימות: 'salesforce', 'hubspot', 'monday', 'pipedrive', 'excel', 'google_workspace', אחר"],
      "integrationNeeds": ["array - צרכי אינטגרציה ספציפיים"],
      "technicalCapability": "high|medium|low - יכולת טכנית של הארגון"
    },
    "roi": {
      "expectedOutcomes": ["array - תוצאות מצופות בעברית"],
      "successMetrics": ["array - מדדי הצלחה"],
      "timeline": "string - ציר זמן ליישום",
      "currentPainCost": "string - עלות הבעיה הנוכחית (שעות/כסף)"
    },
    "buyingSignals": {
      "urgency": "high|medium|low",
      "decisionMaker": boolean - האם מדבר עם מקבל ההחלטות,
      "budgetApproved": boolean - האם יש תקציב מאושר,
      "competitorsMentioned": ["array - מתחרים שהוזכרו"],
      "timelineToDecision": "string - זמן משוער להחלטה"
    }
  },
  "recommendedSolutions": [
    {
      "service": "string - שם השירות מתוך הידע הקיים",
      "priceRange": "string - טווח מחירים מתאים מהמחירון",
      "reasoning": "string - הסבר בעברית למה פתרון זה מתאים"
    }
  ],
  "nextSteps": [
    "array - צעדים הבאים ממוקדים ומעשיים בעברית"
  ],
  "missingInformation": [
    "array - מידע קריטי שחסר לצורך הצעת מחיר מדויקת"
  ]
}
</json_output_schema>

**CRITICAL RULES FOR JSON OUTPUT:**
1. Return ONLY the JSON object - no markdown, no code blocks, no explanatory text
2. ALL field names must be in English (as shown above)
3. ALL content values should be in Hebrew unless they are enums/codes
4. ONLY include fields where you have HIGH confidence - omit uncertain fields entirely
5. Do NOT use null values - simply omit the field
6. Arrays can be empty [] if no information exists
7. Enums must use EXACT values shown (e.g., "high" not "High" or "גבוה")

<confidence_calibration>
Examples of confidence levels:

HIGH CONFIDENCE - EXTRACT:
- "אנחנו מקבלים כ-150 לידים בחודש דרך הטפסים באתר" → leadVolume: "150 לידים חודשיים"
- "יש לנו 8 אנשים בצוות המכירות" → salesTeamSize: "8"
- "אנחנו עובדים עם Salesforce" → currentSystems: ["salesforce"]

MEDIUM CONFIDENCE - DO NOT EXTRACT:
- "אנחנו מקבלים הרבה לידים" → Too vague, omit leadVolume
- "הצוות שלנו לא כל כך גדול" → Unclear, omit teamSize

LOW CONFIDENCE - DO NOT EXTRACT:
- Customer mentions "startup" → Don't assume budget or employee count
- "זה דחוף" → Don't assume specific timeline without clarification
</confidence_calibration>

<solution_matching_logic>
When generating recommendedSolutions, follow this logic:

STEP 1 - SELECT BASE PACKAGE:
1. Lead volume + manual handling → Business Automation Workflow (₪700)
2. Customer inquiries + needs chatbot → AI Agent Base Package (₪1,000)
3. E-commerce store → E-commerce Intelligence (₪3,500, includes AI agent)
4. Need to connect systems → Integration Package (₪1,000)
5. No CRM/using Excel → System Implementation (₪2,500)
6. Content creation needs → Content & Graphics AI (₪3,500)

STEP 2 - ADD RELEVANT ADD-ONS (examples):
- WhatsApp communication needed → +WhatsApp Business API Setup (₪1,000)
- Knowledge base/FAQ → +Knowledge Base (₪800)
- Calendar booking → +Calendar Integration (₪500)
- Voice AI → +Voice AI Capabilities (₪2,750)
- Lead qualification logic → +Complex Business Logic (₪1,150)
- Multiple systems sync → +Multi-System Sync (₪2,250)
- Data migration → +Data Migration Service (₪2,500)
- Training needed → +Training & Workshops (₪1,800)

STEP 3 - OR USE PRE-BUILT SERVICE (if exact match):
If customer need matches a specific service from the catalog:
- AI Lead Qualifier → Full service ₪7,000-10,000
- AI Customer Support → Full service ₪8,000-12,000
- WhatsApp Business Integration → Full service ₪2,000-4,000
- CRM Migration → Full service ₪3,000-6,000

PRICING TIERS:
- Simple (₪700-2,000): Base package only or base + 1 simple addon
- Standard (₪2,000-5,000): Base + 2-4 addons
- Advanced (₪5,000-10,000): Base + complex addons or pre-built service
- Enterprise (₪10,000-22,000+): Multi-agent, custom development, white-label

Match solution complexity to business size and budget indicators.
</solution_matching_logic>

<summary_guidelines>
**CRITICAL: The summary field is the MOST IMPORTANT part of your output.**

Your summary must be comprehensive and detailed (10-15 sentences minimum, more if needed). Include ALL of the following:

1. **Who is the contact?**
   - Name (if mentioned), role, background
   - Example: "דום, מפתח עסקי שחזר לאחרונה מארה\"ב לאחר 6 שנים"

2. **What is their business situation?**
   - Current business model, clients they serve, team size
   - Example: "הוא מייצג מספר לקוחות כולל מרפאות שיניים ומסעדה, ומתחיל להביא לקוחות בשבועיים האחרונים"

3. **What are their specific needs/pain points?**
   - Detailed description of each challenge mentioned
   - Example: "הלקוח של דום (מסעדה) מעוניין בצ'אט בוט קולי לשירות לקוחות. למרפאות השיניים הוא צריך אוטומציה לתזכורות ווטסאפ וניהול פגישות"

4. **What solutions were discussed?**
   - Services mentioned, pricing discussed, technical details
   - Example: "דובר על מחיר של ₪2,500-5,000 לפרויקט, בהתאם למורכבות ההטמעה. הוזכרה בעיית האינטגרציה עם תוכנת Rapid"

5. **Technical environment and constraints:**
   - Current systems, integration challenges, technical limitations
   - Example: "מרפאות השיניים משתמשות ב-Rapid (לא Rapid One), שלא תומך ב-API. נדון בפתרון יצירתי דרך אימיילים או ממשקים אחרים"

6. **Relationship and business opportunity:**
   - Type of relationship (partner/client), urgency, decision timeline
   - Example: "דום מעוניין בשיתוף פעולה כשותף עסקי עם מבנה עמלות. הזמין להצטרף לפגישות עם לקוחות (מחר ובשבוע הבא)"

7. **Next meeting/timeline:**
   - Specific dates, commitments made
   - Example: "יש לו פגישה עם מרפאת שיניים מחר ועוד שתיים ביום ראשון. הציע להצטרף אליו בזום לפגישות"

8. **Key insights and strategic notes:**
   - Important points for follow-up, potential objections addressed, competitive mentions
   - Example: "דום השווה ל-VoiceSpin. הוסבר לו שהטכנולוגיה זהה והיתרון שלנו בהתאמה אישית. הדגיש את המגבלות של Voice AI בעברית"

**Format instructions:**
- Write in natural Hebrew, flowing paragraphs
- Use specific details from the conversation (names, numbers, dates)
- Maintain chronological flow when relevant
- Include direct quotes when impactful
- Be thorough but organized - group related information together
- Aim for 200-400 words in Hebrew

**Example of a GOOD detailed summary:**
"שיחת היכרות עם דוד (דום) לוי, מפתח עסקי בעל ניסיון של 6 שנים בארה\"ב (ניו יורק ופלורידה) שחזר לאחרונה לארץ ומתחיל להביא לקוחות בתחום האוטומציות וה-AI. דום מייצג מספר לקוחות כולל מרפאות שיניים ומסעדה, והוא מחפש ספק טכני אמין לביצוע פרויקטים. העניין המרכזי בשיחה: מסעדה שמעוניינת בצ'אט בוט קולי, ומרפאות שיניים שצריכות אוטומציה לתזכורות ווטסאפ וניהול פגישות. דובר על טווח מחירים של ₪2,500-5,000 לפרויקט, בהתאם למורכבות ההטמעה והכלים שיידרשו. הוזכרה בעיה מרכזית: מרפאות השיניים משתמשות בתוכנת Rapid (לא Rapid One) שלא תומכת ב-API, ונדונו פתרונות יצירתיים כמו אוטומציה דרך אימיילים. דום הציג עצמו כמי שמתחיל לפעול בשוק - כבר סגר כמה לקוחות ויש לו פגישות מתוכננות מחר ובשבוע הבא. הוא הביע עניין ברור בשיתוף פעולה כשותף עסקי עם מבנה עמלות על לקוחות שיסגור, והזמין להצטרף אליו לפגישות הקרובות בזום (במילואים שבוע הבא אז ממילא יהיה דרך זום). בשיחה הוזכר VoiceSpin כמתחרה, והוסבר לדום שהטכנולוגיה זהה אבל היתרון שלנו בהתאמה אישית ובגמישות כחברה קטנה (3 עובדים). הודגש שVoice AI בעברית עדיין לא מושלם ויש מגבלות שצריך להיות שקוף לגביהן. דום הראה מוכנות טכנית גבוהה - הוא מכיר כלים כמו Claude Code, Cursor, N8N, ומבין היטב את עולם ה-AI והאוטומציות. נדונו גם העדפות CRM - Zoho על פני Monday בשל הגמישות והמודולריות. האווירה בשיחה הייתה חיובית ושיתופית, עם תחושה ברורה של potential לשותפות ארוכת טווח."

**Example of a BAD summary (too short):**
"שיחה עם דום, מפתח עסקי. הוא מחפש פתרונות אוטומציה ללקוחות שלו. דובר על מחירים ועל שיתוף פעולה."

Your summary should be closer to the GOOD example - detailed, specific, and comprehensive.
</summary_guidelines>

<next_steps_guidelines>
Generate specific, actionable next steps such as:
✓ "לקבוע פגישת זום של 30 דקות להדגמת סוכן AI למכירות"
✓ "לשלוח דוגמאות של דשבורדים אוטומטיים למעקב לידים"
✓ "להבהיר את תקציב הפרויקט והציר זמן ליישום"
✓ "לבדוק אפשרות אינטגרציה עם המערכת הקיימת שלהם (Monday.com)"

✗ Avoid vague steps like: "להמשיך בתהליך", "לעדכן בהמשך"
</next_steps_guidelines>

<edge_case_handling>
- If transcript is in English: Extract in English, but note in summary: "שיחה התקיימה באנגלית"
- If multiple contradictory statements: Extract the most recent/clear statement, note contradiction in missingInformation
- If customer asks only general questions without revealing needs: Set confidence to "low", focus summary on their questions
- If technical jargon is unclear: Include in missingInformation field for clarification
- If budget is "no budget mentioned": Omit budget field entirely, add to missingInformation
</edge_case_handling>

<quality_checklist>
Before returning your JSON, verify:
□ JSON is valid and parseable (no trailing commas, proper quotes)
□ **Summary is DETAILED and COMPREHENSIVE (10-15 sentences minimum)**
□ Summary includes all 8 required components (who, what, needs, solutions, technical, relationship, timeline, insights)
□ Summary uses specific details from the conversation (names, numbers, dates, systems mentioned)
□ All extracted information has HIGH confidence
□ Hebrew content is properly encoded
□ Enum values match exactly (e.g., "high" not "High")
□ recommendedSolutions reference actual אוטומציות.AI services
□ Price ranges match the company's pricing structure
□ nextSteps are specific and actionable
□ No fields included with null or "unknown" - omit instead
</quality_checklist>`;

    const userPrompt = `Analyze this conversation transcript and extract business information for client ID: ${clientId}\n\n${transcript}`;

    // Call Claude API
    const message = await anthropic.messages.create({
      model: 'claude-sonnet-4-5-20250929',
      max_tokens: 8192,
      temperature: 0.3,
      system: systemPrompt,
      messages: [
        {
          role: 'user',
          content: userPrompt
        }
      ]
    });

    // Extract the response text
    const responseText = message.content[0].text;

    // Try to parse the JSON response
    let analysisResult;
    try {
      const jsonMatch = responseText.match(/\{[\s\S]*\}/);
      if (jsonMatch) {
        analysisResult = JSON.parse(jsonMatch[0]);
      } else {
        analysisResult = JSON.parse(responseText);
      }
    } catch (parseError) {
      console.error('[Custom Transcription] Failed to parse Claude response as JSON:', responseText);
      return res.status(500).json({
        error: 'Failed to parse analysis result',
        rawResponse: responseText
      });
    }

    // Validate the response structure
    if (!analysisResult.extractedFields) {
      return res.status(500).json({
        error: 'Invalid analysis result structure',
        rawResponse: responseText
      });
    }

    // Return the analysis result directly to the calling HTTP node
    res.json({
      success: true,
      message: 'Transcript analyzed successfully',
      clientId,
      transcript,
      summary: analysisResult.summary || 'לא זמין',
      confidence: analysisResult.confidence || 'medium',
      extractedFields: analysisResult.extractedFields,
      recommendedSolutions: analysisResult.recommendedSolutions || [],
      nextSteps: analysisResult.nextSteps || [],
      missingInformation: analysisResult.missingInformation || [],
      rawAnalysis: responseText,
      timestamp: new Date().toISOString(),
      source: 'external-transcription-custom'
    });

  } catch (error) {
    console.error('[Custom Transcription] Error:', error);
    res.status(500).json({
      error: 'Analysis failed',
      message: error.message
    });
  }
}
