// discovery-assistant/scripts/smart-validation.js
#!/usr/bin/env node

/**
 * Smart Validation Script for requirementsPrefillEngine.ts
 * Provides contextual fixes instead of automatic replacements
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

class SmartValidator {
  constructor() {
    this.filePath = path.join(__dirname, '../src/utils/requirementsPrefillEngine.ts');
    this.issues = [];
    this.content = '';
  }

  async validate() {
    console.log('ðŸ” Running smart validation...\n');

    this.content = fs.readFileSync(this.filePath, 'utf8');
    this.findIssues();
    this.displayIssues();
    this.suggestFixes();

    return this.issues;
  }

  findIssues() {
    // 1. ×ž×¦× ×¤×•× ×§×¦×™×•×ª ×©×ž×§×‘×œ×•×ª modules ×›×¤×¨×ž×˜×¨
    const functionPattern = /const (\w+) = \(\s*modules:\s*Meeting\['modules'\]/g;
    const functions = [];

    let match;
    while ((match = functionPattern.exec(this.content)) !== null) {
      functions.push({
        name: match[1],
        startIndex: match.index
      });
    }

    // ×¢×‘×•×¨ ×¢×œ ×›×œ ×¤×•× ×§×¦×™×” ×•×—×¤×© ×©×™×ž×•×©×™× ×©×œ meeting.modules
    functions.forEach(func => {
      const functionBodyPattern = new RegExp(`const ${func.name}.*?^}`, 'ms');
      const functionBodyMatch = this.content.substring(func.startIndex).match(functionBodyPattern);

      if (functionBodyMatch) {
        const functionBody = functionBodyMatch[0];
        const meetingModulesMatches = functionBody.match(/meeting\.modules/g);

        if (meetingModulesMatches) {
          meetingModulesMatches.forEach(match => {
            const matchIndex = functionBody.indexOf(match) + func.startIndex;

            this.issues.push({
              type: 'meeting-modules-usage',
              functionName: func.name,
              lineNumber: this.getLineNumber(matchIndex),
              column: this.getColumnNumber(matchIndex),
              context: this.getContext(matchIndex, 50),
              explanation: this.getExplanation(func.name),
              fix: this.getFix(func.name, match)
            });
          });
        }
      }
    });

    // 2. ×—×¤×© ×©×™×ž×•×©×™× ×©×œ console.log ×‘×ž×§×•× logger
    const consoleLogPattern = /console\.log\(/g;
    let consoleMatch;
    while ((consoleMatch = consoleLogPattern.exec(this.content)) !== null) {
      this.issues.push({
        type: 'console-log-usage',
        lineNumber: this.getLineNumber(consoleMatch.index),
        column: this.getColumnNumber(consoleMatch.index),
        context: this.getContext(consoleMatch.index, 30),
        explanation: '×ž×•×ž×œ×¥ ×œ×”×©×ª×ž×© ×‘-consoleLogger ×‘×ž×§×•× console.log ×™×©×™×¨ ×œ×™×¢×™×œ×•×ª ×¨×™×©×•× ××—×™×“×”',
        fix: {
          original: 'console.log(',
          corrected: 'logger.info(',
          reason: '×”×—×œ×¤×ª console.log ×‘-logger.info ×œ×©×™×ž×•×© ×‘×ž×¢×¨×›×ª ×”×¨×™×©×•× ×”×ž××•×—×“×ª'
        }
      });
    }

    // 3. ×—×¤×© ×©×™×ž×•×©×™× ×©×œ any ×‘×ž×§×•× ×˜×™×™×¤×™× ×¡×¤×¦×™×¤×™×™×
    const anyTypePattern = /:\s*any/g;
    let anyMatch;
    while ((anyMatch = anyTypePattern.exec(this.content)) !== null) {
      this.issues.push({
        type: 'any-type-usage',
        lineNumber: this.getLineNumber(anyMatch.index),
        column: this.getColumnNumber(anyMatch.index),
        context: this.getContext(anyMatch.index, 40),
        explanation: '×ž×•×ž×œ×¥ ×œ×”×’×“×™×¨ ×˜×™×™×¤×™× ×¡×¤×¦×™×¤×™×™× ×‘×ž×§×•× any ×œ×©×™×¤×•×¨ ×‘×˜×™×—×•×ª ×”×˜×™×™×¤×™×',
        fix: {
          original: ': any',
          corrected: ': unknown ××• ×˜×™×™×¤ ×¡×¤×¦×™×¤×™',
          reason: '×”×—×œ×¤×ª any ×‘×˜×™×™×¤ ×¡×¤×¦×™×¤×™ ×™×•×ª×¨ ×œ×‘×˜×™×—×•×ª ×˜×™×™×¤×™× ×ž×©×•×¤×¨×ª'
        }
      });
    }
  }

  getLineNumber(index) {
    const beforeContent = this.content.substring(0, index);
    return beforeContent.split('\n').length;
  }

  getColumnNumber(index) {
    const beforeContent = this.content.substring(0, index);
    const lines = beforeContent.split('\n');
    const lastLine = lines[lines.length - 1];
    return lastLine.length + 1;
  }

  getContext(index, radius = 50) {
    const start = Math.max(0, index - radius);
    const end = Math.min(this.content.length, index + radius);
    return this.content.substring(start, end).replace(/\n/g, ' â†µ ');
  }

  getExplanation(functionName) {
    return `×”×¤×•× ×§×¦×™×” ${functionName} ×ž×§×‘×œ×ª ××ª ×”×¤×¨×ž×˜×¨ 'modules' ××‘×œ ×ž×©×ª×ž×©×ª ×‘-meeting.modules.
            ×–×” ×©×’×•×™ ×›×™ ×”×¤×•× ×§×¦×™×” ×œ× ×ž×§×‘×œ×ª ××ª ×”××•×‘×™×™×§×˜ meeting ×›×¤×¨×ž×˜×¨, ×¨×§ ××ª modules.
            ×¦×¨×™×š ×œ×”×©×ª×ž×© ×‘-modules ×©×”×•×¢×‘×¨ ×›×¤×¨×ž×˜×¨ ×‘×ž×§×•× ×‘-meeting.modules.`;
  }

  getFix(functionName, problematicCode) {
    return {
      original: problematicCode,
      corrected: problematicCode.replace('meeting.modules', 'modules'),
      reason: `×”×—×œ×¤×ª meeting.modules ×‘-modules ×›×™ ×”×¤×•× ×§×¦×™×” ${functionName} ×ž×§×‘×œ×ª ×¨×§ ××ª modules ×›×¤×¨×ž×˜×¨`
    };
  }

  displayIssues() {
    if (this.issues.length === 0) {
      console.log('âœ… ×œ× × ×ž×¦××• ×‘×¢×™×•×ª ×‘×§×•×‘×¥!');
      return;
    }

    console.log(`ðŸš¨ × ×ž×¦××• ${this.issues.length} ×‘×¢×™×•×ª:\n`);

    this.issues.forEach((issue, index) => {
      console.log(`${index + 1}. ×‘×¢×™×” ×‘×¤×•× ×§×¦×™×”: ${issue.functionName}`);
      console.log(`   ×©×•×¨×”: ${issue.lineNumber}, ×¢×ž×•×“×”: ${issue.column}`);
      console.log(`   ×§×˜×¢ ×§×•×“: ${issue.context}`);
      console.log(`   ×”×¡×‘×¨: ${issue.explanation}`);
      console.log(`   ×ª×™×§×•×Ÿ ×ž×•×¦×¢: ${issue.fix.original} â†’ ${issue.fix.corrected}`);
      console.log(`   ×¡×™×‘×”: ${issue.fix.reason}\n`);
    });
  }

  suggestFixes() {
    if (this.issues.length === 0) return;

    console.log('ðŸ’¡ ×”×¦×¢×•×ª ×œ×ª×™×§×•×Ÿ:');
    console.log('');

    this.issues.forEach((issue, index) => {
      console.log(`${index + 1}. ×¢×‘×•×¨ ×”×¤×•× ×§×¦×™×” ${issue.functionName}:`);
      console.log('   ```typescript');
      console.log(`   // ×‘×ž×§×•×:`);
      console.log(`   ${issue.fix.original}`);
      console.log('   ');
      console.log(`   // ×”×©×ª×ž×© ×‘:`);
      console.log(`   ${issue.fix.corrected}`);
      console.log('   ```');
      console.log('');
    });
  }

  async applyFixes(auto = false) {
    if (this.issues.length === 0) {
      console.log('âœ… ××™×Ÿ ×ª×™×§×•× ×™× ×œ×”×—×™×œ');
      return;
    }

    if (!auto) {
      const readline = require('readline');
      const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
      });

      console.log('â“ ×”×× ×œ×”×—×™×œ ××ª ×›×œ ×”×ª×™×§×•× ×™× ×”××•×˜×•×ž×˜×™×™×? (y/n)');
      const answer = await this.question(rl, '');

      if (answer.toLowerCase() !== 'y') {
        console.log('×ª×™×§×•× ×™× ×œ× ×”×•×—×œ×•');
        rl.close();
        return;
      }

      rl.close();
    }

    let newContent = this.content;

    // ×”×—×œ ×ª×™×§×•× ×™× ×ž×”×¡×•×£ ×œ×”×ª×—×œ×” ×›×“×™ ×œ× ×œ×©× ×•×ª ××™× ×“×§×¡×™×
    const sortedIssues = [...this.issues].sort((a, b) => b.lineNumber - a.lineNumber);

    sortedIssues.forEach(issue => {
      const oldCode = issue.fix.original;
      const newCode = issue.fix.corrected;
      newContent = newContent.replace(oldCode, newCode);
    });

    fs.writeFileSync(this.filePath, newContent);
    console.log(`âœ… ×”×•×—×œ×• ${this.issues.length} ×ª×™×§×•× ×™× ×‘×”×¦×œ×—×”!`);
  }

  question(rl, query) {
    return new Promise(resolve => {
      rl.question(query, resolve);
    });
  }
}

// ×”×¤×¢×œ×”
async function main() {
  const validator = new SmartValidator();
  await validator.validate();

  if (process.argv.includes('--fix')) {
    await validator.applyFixes(process.argv.includes('--auto'));
  }
}

export { SmartValidator };
